src/classes/RD2_ApiService.cls

```diff
@@ -0,0 +1,178 @@
+     * class representation of the RdSchedule object because it's protected and customers will not be able to
+     * deserialize it.
+     * @param rdIds Recurring Donation Id's to retrieve schedules for
+     * @return JSON Serialized version of Map<Id, List<RD2_ApiService.Schedule>> schedulesByRdId
+     */
+    public String querySchedules(Set<Id> rdIds) {
+        Map<Id, List<RD2_ApiService.Schedule>> schedulesByRdId = new Map<Id, List<RD2_ApiService.Schedule>>();
+
+        if (!RD2_EnablementService.isRecurringDonations2Enabled) {
+            return JSON.serialize(schedulesByRdId);
+        }
+
+        RD2_ScheduleService.PauseScheduleHandler pauseHandlerService = new RD2_ScheduleService.PauseScheduleHandler();
+
+        String soql = getRecurringDonationQuery();
+        for (npe03__Recurring_Donation__c rd : Database.query(soql)) {
+
+            Boolean isRdCurrentlyPaused = pauseHandlerService.hasActivePause(
+                rd.RecurringDonationSchedules__r, RD2_ScheduleService.currentDate
+            );
+
+            List<RD2_ApiService.Schedule> schedules = new List<RD2_ApiService.Schedule>();
+            for (RecurringDonationSchedule__c s : rd.RecurringDonationSchedules__r) {
+                RD2_ApiService.Schedule schedule = new RD2_ApiService.Schedule(s, rd);
+                if (isRdCurrentlyPaused && schedule.isActiveAndNotPaused()) {
+                    // If the RD is in a pause status, and the schedule record is active & current,
+                    // but not paused, then mark it as not current
+                    schedule.isCurrent = false;
+                }
+                schedules.add(schedule);
+            }
+
+            schedulesByRdId.put(rd.Id, schedules);
+        }
+
+        return JSON.serialize(schedulesByRdId);
+    }
+
+    /**
+     * @return SOQL of Recurring Donation records with a subquery on schedule records
+     */
+    private String getRecurringDonationQuery() {
+        Set<String> queryFields = new Set<String>{
+            'Id',
+            'EndDate__c'
+        };
+        if (UserInfo.isMultiCurrencyOrganization()) {
+            queryFields.add('CurrencyIsoCode');
+        }
+
+        queryFields.add(getScheduleSubQuery());
+
+        String soql = new UTIL_Query()
+            .withFrom(npe03__Recurring_Donation__c.SObjectType)
+            .withSelectFields(queryFields)
+            .withWhere('Id IN :rdIds')
+            .build();
+
+        return soql;
+    }
+
+    /**
+     * @description Returns query on Recurring Donation Schedule
+     * Return all schedules for the RD Schedule visualization.
+     * The Schedule Service will filter them out if not applicable within the time period.
+     * @return String RD Schedule subquery used in the Recurring Donation SOQL
+     */
+    public String getScheduleSubQuery() {
+        Set<String> fields = new RD2_ScheduleService.ScheduleQueryHandler().getFields();
+        fields.add('CreatedDate');
```

Comment:
I apologize for not thinking of this earlier, but would it make some sense to add ModifiedDate as well?  

I was originally thinking we did not need it because we don't modify schedules, but we do.  When we deactivate a schedule, we do so by changing the EndDate__c to today()-1, so ModifiedDate would show us when that happened.  

We can usually infer that from StartDate__c of the new schedule, so it isn't critical that we have this, but it would be a bit more transparent I think.  And we might encounter some scenarios with re-activated RDs where it would be useful to have.